### PR TITLE
Update WR (pixel snapping improvements).

### PR DESCRIPTION
**Pull request by glennw**
This includes a WR update that improves pixel snapping. Specifically,
the non-transform shader now snaps to device pixels in world space,
rather than local space.

This makes 31 new tests pass, and adds 4 new failures.

Of the new failures:
 - The three background size tests appear to be an OSMesa accuracy
   issue. Near the bottom of the image, a small number of pixels
   are different from the reference by a value of 1.0 / 255.0. I've
   confirmed on 3 real GPUs and also on llvmpipe that the tests
   match on those devices. This would be a good case for fuzzy test
   support in WPT.
 - The other failing test appears to have unreasonable accuracy
   requirements as it also fails on Gecko and Chromium in the same
   way that the test now fails on WR.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/15467)

### CODE REVIEW

**Comment by glennw** [comment]
@bors-servo try
 * Let's see if the test expectation changes on my local machine match on CI.

**Comment by emilio** [comment]
```
  ▶ PASS [expected FAIL] /_mozilla/css/border_radius_elliptical_a.html

  ▶ FAIL [expected PASS] /_mozilla/css/inline_block_overflow_hidden_a.html
  └   → /_mozilla/css/inline_block_overflow_hidden_a.html 7cbcaad2862c9c8a9cc7c4e2f28d727a4db30cdf
/_mozilla/css/inline_block_overflow_hidden_ref.html 84f6e84616988e3389b2929cd7f5617e4d61668b
Testing 7cbcaad2862c9c8a9cc7c4e2f28d727a4db30cdf == 84f6e84616988e3389b2929cd7f5617e4d61668b

  ▶ FAIL [expected PASS] /_mozilla/css/input_selection_a.html
  └   → /_mozilla/css/input_selection_a.html 88bbf73124d6e3403177356b5b52c36d191af4d5
/_mozilla/css/input_selection_ref.html 7d438f12cb00b19560a952fb74769e8bf93baacd
Testing 88bbf73124d6e3403177356b5b52c36d191af4d5 == 7d438f12cb00b19560a952fb74769e8bf93baacd

  ▶ FAIL [expected PASS] /_mozilla/css/position_fixed_a.html
  └   → /_mozilla/css/position_fixed_a.html cdda74fb1d41a76d413cdd5259ec00200509eb22
/_mozilla/css/position_fixed_b.html bf4079efd49f6860c34e45048b8cecca2e4f5b72
Testing cdda74fb1d41a76d413cdd5259ec00200509eb22 == bf4079efd49f6860c34e45048b8cecca2e4f5b72

  ▶ FAIL [expected PASS] /_mozilla/mozilla/simple_scroll_to_fragment.html
  └   → /_mozilla/mozilla/simple_scroll_to_fragment.html 82a5021224ffa04f7b5073a336901f8269fb9591
/_mozilla/mozilla/simple_scroll_to_fragment_ref.html f88bdab49fc90a67147c387dbf35283341c5e0da
Testing 82a5021224ffa04f7b5073a336901f8269fb9591 == f88bdab49fc90a67147c387dbf35283341c5e0da
```

**Comment by glennw** [comment]
@bors-servo retry
 * Let's see if those are intermittent or permanent. In the meantime I'll start trying to reproduce those locally on mac.

**Comment by nox** [comment]
I put your commit in https://github.com/servo/servo/pull/15479 to see if it makes the failure disappear with the rounding fixes from app_units.

**Comment by nox** [comment]
This is currently blocking the serde bump.

**Comment by nox** [comment]
It also doesn't bump WR to the latest version, but prior to @nical's changes to use a thread pool.

**Comment by glennw** [comment]
Rebased and updated to master of WR. However, I believe there are still some mac-specific regressions that will need a PR to land in WR today before this is ready.

**Comment by glennw** [comment]
@bors-servo retry
 * Testing if those Linux fails are intermittents, and checking the mac regressions.

**Comment by glennw** [comment]
@bors-servo try
 * Trying once more to see if these weird crashes / failures are related to the WR update.

**Comment by glennw** [comment]
@nox OK, I know you're waiting on this to land - so here's an update on the current status:

* When I opened the PR, it became apparent there were a few regressions that were occurring on the mac CI only (not on my local mac) due to the pixel snapping changes in WR.
* After rebasing this PR, it appears that most (all?) of those tests that had regressed have been disabled in Servo in the meantime (by your app_units update PR).
* Unfortunately, the CI logs then started showing a heap of seemingly unrelated crashes and failures on a try run. I checked locally and with the WR update, my test runs are consistently 5-10% *faster* than without the WR update - it's possible this is exposing some existing race conditions / bugs in Servo?
* In the meantime, I have been working on a pixel snapping follow up PR - https://github.com/servo/webrender/pull/870 - this fixes several more tests, and also fixes most (all?) of the tests you had to disable with the app_units update.

So, in summary, the major issue holding this PR up is the seemingly unrelated crashes that are occurring on CI machines with this update, if you want to investigate those. Ideally, we'd also get https://github.com/servo/webrender/pull/870 merged and included in this update, since it is a significant improvement and allows us to re-enable several reftests. So far, I've been unable to reproduce any of those strange crashes locally - the WPT/CSS tests run fine on my machine.

**Comment by glennw** [comment]
@bors-servo try
 * Re-enabled some of the tests that were disabled with the app_units update.

**Comment by glennw** [comment]
@bors-servo try
 * Updated some more test expectations, the other 3 failures listed are known intermittents.

**Comment by glennw** [comment]
@bors-servo try
 * Update some mac expectations (these are ones that are fixed by the pending follow up snapping PR I mentioned above).

**Comment by glennw** [comment]
@bors-servo try
 * Name the expectation files properly...
